Wikipedia/Code/WMFSettingsViewController.m

```diff
@@ -382,16 +382,48 @@ - (void)showClearCacheActionSheet {
     message = [NSString localizedStringWithFormat:message, bytesString];
 
     UIAlertController *sheet = [UIAlertController alertControllerWithTitle:WMFLocalizedStringWithDefaultValue(@"settings-clear-cache-are-you-sure-title", nil, nil, @"Clear cached data?", @"Title for the confirmation presented to the user to verify they are sure they want to clear clear cached data.") message:message preferredStyle:UIAlertControllerStyleAlert];
+    typeof(self) __weak weakSelf = self;
     [sheet addAction:[UIAlertAction actionWithTitle:WMFLocalizedStringWithDefaultValue(@"settings-clear-cache-ok", nil, nil, @"Clear cache", @"Confirm action to clear cached data")
                                               style:UIAlertActionStyleDestructive
                                             handler:^(UIAlertAction *_Nonnull action) {
-                                                [self.dataStore clearTemporaryCache];
+                                                [weakSelf clearCache];
                                             }]];
     [sheet addAction:[UIAlertAction actionWithTitle:WMFLocalizedStringWithDefaultValue(@"settings-clear-cache-cancel", nil, nil, @"Cancel", @"Cancel action to clear cached data {{Identical|Cancel}}") style:UIAlertActionStyleCancel handler:NULL]];
 
     [self presentViewController:sheet animated:YES completion:NULL];
 }
 
+- (void)clearCache {
+    
+    [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(showClearCacheInProgressBanner) object:nil];
+    [self performSelector:@selector(showClearCacheInProgressBanner) withObject:nil afterDelay:1.0];
```

Comment:
Good, glad you haven't seen the double banners!

Yeah, this may be me going overboard on performance testing, but I (programmatically) loaded a year's worth of Explore feed data and tested this on my old iPhone 7. It caused an alarming UI freeze in this situation, which is why I added [4c244cb](https://github.com/wikimedia/wikipedia-ios/pull/4574/commits/4c244cb5f4573ddd35562e2f0c7c543fa9dc96ed) to perform the cleanup on a different thread and show banners.

So if something will take a while, the user will see that a cleanup started via the banner, but the UI is still responsive. They can then go back to the Explore feed. Once cleanup completes, the Explore feed refreshes and the "complete" banner displays.